content/en/docs/guides/retry.md

```diff
@@ -0,0 +1,129 @@
+---
+title: Retry 
+description: >-
+  gRPC takes the stress out of failures! Get fine-grained retry control and detailed insights with OpenCensus and OpenTelemetry support.
+---
+
+### Overview
+
+Retries are a key pattern for making services more reliable. By re-attempting failed operations, applications can overcome temporary issues like network or server glitches. This is essential for modern cloud applications to handle the inevitable transient faults that occur.
+
+For the best practice, applications should understand what failed operations are suitable for retry, define exponential backoff parameters for retry delay, determine the number of retry attempts, and also monitor retry metrics.
+
+
+### How gRPC client retry works
+
+gRPC’s built-in retry logic saves the call's history for potential retries and monitors RPC events. Even if there is no retry policy configured, gRPC still saves the call's history in case it needs to perform transparent retry (discussed in a later section). Note that ‘retry’ means replacing a failed call with a new call and replaying the call's history on that newly created call.
+
+If certain criteria are met – the RPC closes with a failure status code matching the retry policy's retryable status codes and remains within the retry attempt limit – gRPC will create a new retry stream after an exponential backoff delay.
+
+gRPC also supports other features like retry throttling and server push back. See [gRFC for client side retry] for further details.
+
+Once the response header is received, the RPC is committed. No further retries will be attempted, and gRPC hands over the RPC to the application.
+
+The graph below shows architectural overview of gRPC retry internal.
+
+
+```mermaid
+sequenceDiagram
+  Application ->> gRPC Client: Configure retry policy. <br> Send request to dns:///my-service
+  gRPC Client ->> Server: Create retry attempt 1 
+  Server -->> gRPC Client : RPC closed with error 
+  gRPC Client ->> Server: Create retry attempt 2 
+  Server -->> gRPC Client: Successful 
+  gRPC Client ->> Application: No more retry. Proceed.
+```
+
+### Retry configuration
+Retries are enabled by default even without specifying a retry policy. You can also disable retries entirely.
```

Comment:
This is a bit confusing to figure out what it means. I know we mention transparent retries in the "How gRPC client retry works" section, but still.

How about something like:
> Retries are enabled by default, but there is no default retry policy. Without a retry policy, gRPC cannot safely retry RPCs in most cases. Only RPCs that failed due to low-level races are retried, and only if gRPC is certain the RPCs have not been processed by a server. This is known as "transparent retry." You can configure a retry policy to allow gRPC to retry RPCs in more circumstances and more aggressively. You can also disable retries entirely when creating a channel, which disables transparent retries and any configured retry policies.